### PR TITLE
[Doc] Add temporary partitions

### PR DESCRIPTION
**Pull request by hellolilyliuyi**
Fixes #issue

## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [X] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [X] I have checked the version labels which the pr will be auto-backported to the target branch
  - [X] 3.0
  - [X] 2.5
  - [X] 2.4
  - [X] 2.3

### CODE REVIEW

File: docs/sql-reference/sql-statements/data-manipulation/CREATE ROUTINE LOAD.md [hunk 1]
```diff
@@ -44,6 +44,7 @@ Optional. The properties of the data. Syntax:
 [COLUMNS (<column1_name>[,<column2_name>,<column_assignment>,... ])],
 [WHERE <expr>],
 [PARTITION (<partition1_name>[,<partition2_name>,...])]
+[TEMPORARY PARTITION (<temporary_partition1_name>[,<temporary_partition2_name>,...])]
```
Comment (amber-create) [inline docs/sql-reference/sql-statements/data-manipulation/CREATE ROUTINE LOAD.md:47]: image
spaces are missing in between comma (,) and the left angle bracket (<)

File: docs/table_design/Temporary_partition.md [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+```SQL
+ALTER TABLE <table_name>
+ADD TEMPORARY PARTITIONS START ("value1") END ("value2") EVERY {(INTERVAL <num> <time_unit>)|<num>}
+[(partition_desc)]
+[DISTRIBUTED BY HASH(<bucket_key>) BUCKETS <bucket_number>];
+```
+
+### Parameters
+
+`partition_desc`:  specifies the number of buckets and properties for temporary partitions, such as the number of replicas and the storage medium.
```
Comment (amber-create) [inline docs/table_design/Temporary_partition.md:51]: image
redundant spaces

File: docs/table_design/Temporary_partition.md [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+### Load data by using STREAM LOAD
+
+Example:
+
+```undefined
+curl --location-trusted -u root: -H "label:123" -H "temporary_partitions: tp1, tp2, ..." -T testData \
+    http://host:port/api/example_db/site_access/_stream_load    
+```
+
+For detailed information such as the syntax and parameters descriptions, see [STREAM LOAD](..sql-reference/sql-statements/data-manipulation/STREAM%20LOAD.md).
```
Comment (amber-create) [inline docs/table_design/Temporary_partition.md:124]: image
See Line 133

File: docs/table_design/Temporary_partition.md [hunk 4]
```diff
@@ -0,0 +1,311 @@
...
+(
+    DATA INFILE("hdfs://hdfs_host:hdfs_port/user/starrocks/data/input/file")
+    INTO TABLE my_table
+    TEMPORARY PARTITION (tp1, tp2, ...)
+    ...
+)
+WITH BROKER "mybroker"
+(
+    "username" = "hdfs_username",
+    "password" = "hdfs_password"
```
Comment (amber-create) [inline docs/table_design/Temporary_partition.md:152]: image
From v2.5, no brokers are required in most scenarios. See Lines 138 to 141 which are correct.

File: docs/table_design/Temporary_partition.md [hunk 5]
```diff
@@ -0,0 +1,311 @@
...
+);
+```
+
+For detailed information such as the syntax and parameters descriptions, see [Broker Load](../sql-reference/sql-statements/data-manipulation/BROKER%20LOAD.md).
```
Comment (amber-create) [inline docs/table_design/Temporary_partition.md:156]: image
See Line 133

File: docs/table_design/Temporary_partition.md [hunk 6]
```diff
@@ -0,0 +1,311 @@
...
+    TEMPORARY PARTITION (tp1, tp2, ...)
+    ...
+)
+WITH BROKER "mybroker"
+(
+    "username" = "hdfs_username",
+    "password" = "hdfs_password"
+);
+```
+
```
Comment (amber-create) [inline docs/table_design/Temporary_partition.md:155]: I think it's better to add a NOTE above or to Line 156 to tell users that StorageCredentialParams is not something they can directly copy and run but a set of authentication parameter that they need to configure based on the authentication method they choose. For example: "Note that StorageCredentialParams represents a group of authentication parameters which vary depending on the authentication method you choose." followed by "For detailed syntax and parameter descriptions, see Broker Load."

File: docs/table_design/Temporary_partition.md [hunk 7]
```diff
@@ -0,0 +1,311 @@
...
+COLUMNS(col, col2,...),
+TEMPORARY PARTITIONS(tp1, tp2, ...)
+FROM KAFKA
+(
+    "kafka_broker_list" ="<kafka_broker1_ip>:<kafka_broker1_port>,<kafka_broker2_ip>:<kafka_broker2_port>",
+    "kafka_topic" = "ordertest"
+);
+```
+
+For detailed information such as the syntax and parameters descriptions, see [CREATE ROUTINE LOAD](../sql-reference/sql-statements/data-manipulation/CREATE%20ROUTINE%20LOAD.md).
```
Comment (amber-create) [inline docs/table_design/Temporary_partition.md:173]: image
See Line 133

File: docs/table_design/Temporary_partition.md [hunk 8]
```diff
@@ -0,0 +1,311 @@
...
+site_access TEMPORARY PARTITION (tp1, tp2, ...);
+```
+
+You can use the JOIN clause to query data in temporary partitions from two tables.
+
+```SQL
+SELECT * FROM
+site_access TEMPORARY PARTITION (tp1, tp2, ...)
+JOIN
+site_access_copy TEMPORARY PARTITION (tp1, tp2, ...)
```
Comment (amber-create) [inline docs/table_design/Temporary_partition.md:196]: image
should be "(<temporary_partition1_name>[, <temporary_partition2_name>, ...)

**Review by amber-create — APPROVED** [review]

**Review by evelynzhaojie — APPROVED** [review]

**Comment by hellolilyliuyi** [comment]
@mergify backport branch-3.0 branch-2.5 branch-2.4 branch-2.3

**Pull request merged**